MetaU Capstone Project/View Controllers/MusicSessionViewController.m

```diff
@@ -44,13 +50,46 @@ - (void)viewDidLoad {
     self.isPlaying = NO;
     
     [self querySetup];
+    
+    [MusicSession addUserToSession:self.musicSession.sessionCode withCompletion:^(BOOL succeeded, NSError * error) {
+        if (error != nil) {
+            NSLog(@"Error: %@", error.localizedDescription);
+        } else {
+            NSLog(@"User added to session successfully");
+        }
+    }];
+    
+    [self getDataFrom:@"https://api.spotify.com/v1/me/player/recently-played"];
+}
+
+- (void)testTimer // Increments counter every second
+{
+    [NSTimer scheduledTimerWithTimeInterval:1.0f
+                                 target:self selector:@selector(testTimestamp:) userInfo:nil repeats:YES];
+}
+
+-  (void)testTimestamp:(NSTimer *)timer {
+
+    self.testCounter++;
+    
+    PFQuery *query = [[MusicSession query] whereKey:@"sessionCode" equalTo:self.musicSession.sessionCode];
+    
+    [query getObjectInBackgroundWithId:self.musicSession.objectId
+                                 block:^(PFObject *session, NSError *error) {
+        PFUser *host = session[@"host"];
+        
+        if ([PFUser.currentUser.objectId isEqual:host.objectId]) {
+            session[@"timestamp"] = @(self.testCounter);
+            [session saveInBackground];
+        }
+    }];
 }
 
 - (void)querySetup {
     self.client = [[PFLiveQueryClient alloc] initWithServer:SERVER_URL applicationId:APPLICATION_ID clientKey:CLIENT_KEY];
     PFQuery *query = [PFQuery queryWithClassName:@"MusicSession"];
     self.subscription = [self.client subscribeToQuery:query];
-
+    
     __unsafe_unretained typeof(self) weakSelf = self;
```

Comment:
In general, use __weak typeof(self) weakSelf instead of __unsafe_unretained since there's some zeroing done when using __unsafe_unretained. You can also use auto __weak weakSelf too!